src/main/java/org/opensearch/performanceanalyzer/reader_writer_shared/EventLogFileHandler.java

```diff
@@ -174,8 +174,10 @@ private void readInternal(Path pathToFile, int bufferSize, EventDispatcher proce
     public void deleteAllFiles() {
         LOG.debug("Cleaning up any leftover files.");
         File root = new File(metricsLocation);
-        // Filter out '.tmp' files, we do not want to delete currBucket .tmp files
-        String[] filesToDelete = root.list((dir, name) -> !name.endsWith(TMP_FILE_EXT));
+        String[] filesToDelete = root.list();
+        if (filesToDelete == null) {
+            return;
+        }
```

Comment:
Do you think we should retain filtering out .tmp files check as this might delete the currentBucket .tmp files?